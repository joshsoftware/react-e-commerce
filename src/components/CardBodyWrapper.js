import React from 'react';
import { CardBody } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardBodyWrapper = ({ body_content, className }) => {
  return <CardBody className={className}>{body_content}</CardBody>;
};

export default CardBodyWrapper;

CardBodyWrapper.propTypes = {
  body_content: PropTypes.array.isRequired,
  className: PropTypes.string
};
