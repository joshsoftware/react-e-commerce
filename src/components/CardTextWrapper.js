import React from 'react';
import { CardText } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardTextWrapper = ({ text, className }) => {
  return <CardText className={className}>{text}</CardText>;
};

export default CardTextWrapper;

CardTextWrapper.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string
};
