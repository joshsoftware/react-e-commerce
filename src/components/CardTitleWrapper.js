import React from 'react';
import { CardTitle } from 'reactstrap';
import { PropTypes } from 'prop-types';

const CardTitleWrapper = ({ title, className }) => {
  return <CardTitle className={className}>{title}</CardTitle>;
};

export default CardTitleWrapper;

CardTitleWrapper.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string
};
