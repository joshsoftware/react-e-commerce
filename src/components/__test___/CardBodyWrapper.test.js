import React from 'react';
import { render } from '@testing-library/react';
import CardTitleWrapper from '../CardTitleWrapper';
import CardTextWrapper from '../CardTextWrapper';
import CardBodyWrapper from '../CardBodyWrapper';

describe('Card Body', () => {
  it('Must Return Card Body', () => {
    let body_content = [];
    body_content.push(<CardTitleWrapper key={'1'} title={'Project Title'} />);
    body_content.push(<CardTextWrapper key={'2'} text={'$1000'} />);

    const { asFragment } = render(<CardBodyWrapper body_content={body_content} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
