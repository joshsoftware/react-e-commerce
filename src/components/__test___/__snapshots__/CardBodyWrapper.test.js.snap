// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Body Must Return Card Body 1`] = `
<DocumentFragment>
  <div
    class="card-body"
  >
    <div
      class="card-title"
    >
      Project Title
    </div>
    <p
      class="card-text"
    >
      $1000
    </p>
  </div>
</DocumentFragment>
`;
